cmd/query/app/querysvc/query_service_test.go

```diff
@@ -59,42 +59,63 @@ var (
 	}
 )
 
-func initializeTestServiceWithArchiveOptions() (*QueryService, *spanstoremocks.Reader, *depsmocks.Reader, *spanstoremocks.Reader, *spanstoremocks.Writer) {
-	readStorage := &spanstoremocks.Reader{}
-	dependencyStorage := &depsmocks.Reader{}
-	archiveReadStorage := &spanstoremocks.Reader{}
-	archiveWriteStorage := &spanstoremocks.Writer{}
-	options := QueryServiceOptions{
-		ArchiveSpanReader: archiveReadStorage,
-		ArchiveSpanWriter: archiveWriteStorage,
+type testMocks struct {
+	spanReader *spanstoremocks.Reader
+	depsReader *depsmocks.Reader
+
+	archiveSpanReader *spanstoremocks.Reader
+	archiveSpanWriter *spanstoremocks.Writer
+}
+
+type QueryServiceOptionApplier func(*testMocks, *QueryServiceOptions)
```

Comment:
usually these types are named simply XyzOption

Reply:
btw, I'm planning to do the same thing for [`QueryServiceOptions`](https://github.com/jaegertracing/jaeger/blob/c5642b708be9e2577cc1c494889ae97946ccc78a/cmd/query/app/querysvc/query_service.go#L41) in the next PR, as I think MetricsQuery reader belongs in this struct. 

Can you think of a good name for the `Option`? `QueryServiceOptionsOption` seems a little too stutter-y and I'm a little short on other ideas on names. :P